internal/main.go

```diff
@@ -80,14 +80,26 @@ func main() {
 		flags.provider = parseCmdline(args)
 	}
 
+	var metadataFn func() (providers.Metadata, error)
 	switch flags.provider {
-	case "azure", "digitalocean", "ec2", "gce", "packet", "openstack-metadata":
+	case "azure":
+		metadataFn = azure.FetchMetadata
+	case "digitalocean":
+		metadataFn = digitalocean.FetchMetadata
+	case "ec2":
+		metadataFn = ec2.FetchMetadata
+	case "gce":
+		metadataFn = gce.FetchMetadata
+	case "packet":
+		metadataFn = packet.FetchMetadata
+	case "openstack-metadata":
+		metadataFn = openstackMetadata.FetchMetadata
```

Comment:
I'd rather see this logic splits in its own testable function, bubbling-up errors. See the general comment for details.